libraries/exoplayer_rtsp/src/test/java/androidx/media3/exoplayer/rtsp/reader/RtpH263ReaderTest.java

```diff
@@ -0,0 +1,198 @@
+/*
+ * Copyright 2022 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package androidx.media3.exoplayer.rtsp.reader;
+
+import static androidx.media3.common.util.Util.getBytesFromHexString;
+import static com.google.common.truth.Truth.assertThat;
+
+import androidx.media3.common.Format;
+import androidx.media3.common.MimeTypes;
+import androidx.media3.common.util.ParsableByteArray;
+import androidx.media3.exoplayer.rtsp.RtpPacket;
+import androidx.media3.exoplayer.rtsp.RtpPayloadFormat;
+import androidx.media3.test.utils.FakeExtractorOutput;
+import androidx.media3.test.utils.FakeTrackOutput;
+import androidx.test.ext.junit.runners.AndroidJUnit4;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.primitives.Bytes;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.junit.MockitoJUnit;
+import org.mockito.junit.MockitoRule;
+
+/**
+ * Unit test for {@link RtpH263Reader}.
+ */
+@RunWith(AndroidJUnit4.class)
+public final class RtpH263ReaderTest {
+  private static final byte[] FRAME_1_FRAGMENT_1_DATA =
+      getBytesFromHexString("80020c0419b7b7d9591f03023e0c37b");
+  private final RtpPacket FRAME_1_FRAGMENT_1 =
```

Comment:
can you make the field `static final`?

Reply:
Since this data gets changed by the reader, making it static will affect other tests.

Reply:
Do you mean that the tests will see different payload data even when they reference `FRAME_1_FRAGMENT_1`?

Like `FRAME_1_FRAGMENT_1` is different in `consume_validPackets` and `consume_outOfOrderPackets`?

Reply:
Yes. After the first test, FRAME_1_FRAGMENT_1 will change. It will be 0000 + frame_1_fragment_1_data + frame_1_fragment_2_data. 
The payload header part will no longer be 0400. It will be 0000.

Reply:
I don't think it's good practice to have that in test then. A reader would expect the test to run independently, like if I see `consume_outOfOrderPackets` references `FRAME_1_FRAGMENT_1`, I expect the payload header to be 0400.

On a more general note, `FRAME_1_FRAGMENT_1` looks like a constant and let's keep it constant. The tests shouldn't have inner dependencies. Imagine one day I move the tests around, and they are not run in the current order.

Reply:
That makes sense. I have changed the reader to keep the input constant. I have updated the test as well. Please have a look

Reply:
Thanks for making the change, I think we can do it another way though. Your current approach involves in copying the all the received RTP packets again, which increases the pressure on the memory system. The newly introduced memory copy is only needed in tests, but is not actually needed during normal operation.

What about implement a method that copies the packets in tests every time it's referenced? Like 
```java
private static RtpPacket copyPacket(RtpPacket packet) {
  RtpPacket.Builder builder =
      new RtpPacket.Builder()
          .setPadding(packet.padding)
          .setMarker(packet.marker)
          .setPayloadType(packet.payloadType)
          .setSequenceNumber(packet.sequenceNumber)
          .setTimestamp(packet.timestamp)
          .setSsrc(packet.ssrc);

  if (packet.csrc.length > 0) {
    builder.setCsrc(Arrays.copyOf(packet.csrc, packet.csrc.length));
  }
  if (packet.payloadData.length > 0) {
    builder.setPayloadData(Arrays.copyOf(packet.payloadData, packet.payloadData.length));
  }
  return builder.build();
}
```